### ISSUE TITLE
Deprecate & remove serializeTargetValueType

### ISSUE THREAD

**Issue by not-napoleon** [elastic/elasticsearch#58135]
In #42949 we removed all use of target value type in the subclasses of `ValuesSourceAggregationBuilder`, but left the serialization format sending a hard-coded null to preserve compatibility.  This ticket tracks the work of removing that null from the serialization format, and the associated logic in VSAB.

**Comment by kaliljoao** [comment]
Hi guys, I would like to work on this issue.

**Comment by not-napoleon** [comment]
@kaliljoao Sounds good, let me know if you need any help! (Sorry for the slow reply - I was out of office last week)

**Comment by kaliljoao** [comment]
> @kaliljoao Sounds good, let me know if you need any help! (Sorry for the slow reply - I was out of office last week)

Hey @not-napoleon, can U help me? I started my work on this issue today and I am confused about what exactly I need to do. Do I need only to remove where null was hard-coded? My doubt was caused because it is just one line of code, if I am not wrong of course.

**Comment by not-napoleon** [comment]
@kaliljoao You're right, it is only a couple of lines of code.  The difficulty is that we need to maintain backwards compatibility and carefully deprecate this method.  You need to detect the target version (we have some tools for this) and adapt the wire format accordingly.  Honestly, this is a tricky issue and not one I'd really recommend for someone not well versed in how Elasticsearch maintains backwards compatibility.  Usually we don't expect community members to deal with that, which is why this wasn't labeled `help wanted`
